osu.Game/Overlays/Music/MusicActionHandler.cs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Input.Bindings;
+using osu.Game.Beatmaps;
+using osu.Game.Input.Bindings;
+using osu.Game.Overlays.OSD;
+
+namespace osu.Game.Overlays.Music
+{
+    /// <summary>
+    /// Handles <see cref="GlobalAction"/> relating to music playback, and displays a <see cref="Toast"/> via the cached <see cref="OnScreenDisplay"/> accordingly.
+    /// </summary>
+    public class MusicActionHandler : Component, IKeyBindingHandler<GlobalAction>
+    {
+        [Resolved]
+        private IBindable<WorkingBeatmap> beatmap { get; set; }
+
+        [Resolved]
+        private MusicController musicController { get; set; }
+
+        [Resolved]
+        private OnScreenDisplay onScreenDisplay { get; set; }
```

Comment:
any reason for dropping the `[CanBeNull]` constraint here after moving from `MusicController`?

Reply:
Hmm, was originally planning that `MusicActionHandler` shouldn't really be used if `OnScreenDisplay` isn't cached, but seems like the class name is a bit misleading to show that, fair, will return back.

Reply:
i don't understand what you're trying to say. i was asking because removing the null is more restrictive what was there before, and i just wanted to ask if you had any idea why null was even allowed. if the `CanBeNull` is not necessary, it should be dropped.

Reply:
back when it was in `MusicController`, it was quite useful to permit null to allow test scenes that need to test `MusicController`-dependent components (e.g. `NowPlayingOverlay`) not having to cache an `OnScreenDisplay`. 

Makes sense to permit null here as well because `OnScreenDisplay` is more like giving a bonus behaviour in this component rather than a required dependency for this to work.

Reply:
i would have really been fine with non-null